orcid-core/src/main/java/org/orcid/core/manager/impl/SalesForceManagerImpl.java

```diff
@@ -436,6 +471,32 @@ public void createContact(Contact contact) {
         salesForceContactsCache.remove(accountId);
     }
 
+    private void createOpportunityContact(Contact contact, String opportunityId) {
+        String accountId = contact.getAccountId();
+        String contactOrcid = contact.getOrcid();
+        if (StringUtils.isBlank(contact.getEmail())) {
+            Email primaryEmail = emailManager.getEmails(contactOrcid).getEmails().stream().filter(e -> e.isPrimary()).findFirst().get();
+            contact.setEmail(primaryEmail.getEmail());
+        }
+
+        List<Contact> existingContacts = salesForceDao.retrieveAllContactsByAccountId(accountId);
```

Comment:
@wjrsimpson I suspect we might be asked to check all SF contacts for matches, but we can roll with this for now.

Reply:
Yep, hoping we can discuss further at some point.